### ISSUE TITLE
KL loss

### ISSUE THREAD

**Issue by zhunzhong07** [sagiebenaim/OneShotTranslation#4]
Hi, thanks for your wonderful work. 

In your code, you directly use 'mu_2 = torch.pow(mu, 2)' to compute the KL loss between encoded feature and N(0, 1).

However, in traditional, we first compute 'mu' and 'logvar' by the encoder, and compute the KL loss by 'KL = -0.5 * torch.sum(1 + logvar - mu.pow(2) - logvar.exp())'.

Can you explain the difference between these two ways for computing the KL loss?

Thank you very much!

**Comment by zhunzhong07** [comment]
Also, I random sample a feature from N(0, 1) and try to use the decoder to generate an image. The generated image is bad (like random noise).

**Comment by sagiebenaim** [comment]
Hi, thank you for your comment.

The KL loss is indeed different. In the original formulation we used the KL loss similarly to what you write. However post paper submission, we found that the formulation used here works just as well. It is a form of denoizing autoencoder and was also used in the UNIT code. 

For your second question, note that we do not require sampling from the latent space (indeed the latent dimension is too large for that). All generated images (Both for domain A or B) are first encoded, and only then decoded, so it is likely that a randomly sampled feature will not produce a good image.
